### PR TITLE
test: move the pretrained model to the same device as current model before assertion

### PR DESCRIPTION
**Pull request by fnhirwa**
## What does this PR do?
The tests for `tests/tests_pytorch/models/test_restore.py::test_load_model_from_checkpoint` Were failing for me on macOS locally and I found out that there is a runtime error due to a comparison of two Tensors from different devices. 
Just moving the pretrained model to the same device as the current model we are comparing with seems to fix the issue.

I thought it was a quick fix no need to open a github issue.
Thanks

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--19668.org.readthedocs.build/en/19668/

### CODE REVIEW

File: tests/tests_pytorch/models/test_restore.py [hunk 1]
```diff
@@ -465,6 +465,7 @@ def test_load_model_from_checkpoint(tmp_path, model_template):
 
     # Ensure that model can be correctly restored from checkpoint
     pretrained_model = model_template.load_from_checkpoint(last_checkpoint)
+    pretrained_model.to(model.device)  # Move pretrained_model to the same device as model
```
Comment (awaelchli) [inline tests/tests_pytorch/models/test_restore.py:468]: I'm ok with this fix. A slightly better way IMO would be to set

```
"accelerator": "cpu",
```

in the `trainer_options` above. The test doesn't need to run on different accelerators.

**Review by awaelchli — APPROVED** [review]

File: tests/tests_pytorch/models/test_restore.py (hunk 1 above)
Comment (fnhirwa) [reply tests/tests_pytorch/models/test_restore.py:468]: yeah makes sense, if the tests only run on CPU adding this option is enough here 👍

**Review by carmocca — APPROVED** [review]
